internal/emlb/emlb.go

```diff
@@ -233,6 +233,113 @@ func (e *EMLB) ReconcileVIPOrigin(ctx context.Context, machineScope *scope.Machi
 	return nil
 }
 
+// DeleteLoadBalancer deletes the Equinix Metal Load Balancer associated with a given ClusterScope.
+func (e *EMLB) DeleteLoadBalancer(ctx context.Context, clusterScope *scope.ClusterScope) error {
+	log := ctrl.LoggerFrom(ctx)
+
+	packetCluster := clusterScope.PacketCluster
+	clusterName := packetCluster.Name
+
+	// Make sure the cluster already has an EMLB ID in its packetCluster annotations, otherwise abort.
+	lbID, exists := packetCluster.Annotations[loadBalancerIDAnnotation]
+	if !exists || (lbID == "") {
+		log.Info("no Equinix Metal Load Balancer found in cluster's annotations, skipping EMLB delete")
+		return nil
+	}
+
+	// See if the EMLB has a Port ID in its packetCluster annotations.
+	lbPortNumber, exists := packetCluster.Annotations[loadBalancerPortNumberAnnotation]
+	if !exists || (lbPortNumber == "") {
+		log.Info("no Equinix Metal Load Balancer Port Number found in cluster's annotations, skipping EMLB delete")
+		return nil
+	}
+
+	log.Info("Deleting EMLB", "Cluster Metro", e.metro, "Cluster Name", clusterName, "Project ID", e.projectID, "Load Balancer ID", lbID)
+
+	// Fetch the Load Balancer object.
+	// Skip if 404, otherwise error
+	lb, err := e.getLoadBalancer(ctx, lbID)
+	if err != nil {
+		log.Error(err, "failed to load the loadbalancer object, cannot proceed with deletion")
+		return err
+	}
+
+	// Get an int version of the listener port number.
+	portNumber, err := strconv.ParseInt(lbPortNumber, 10, 32)
+	if err != nil {
+		log.Error(err, "failed to convert the loadbalancer port number to an int, cannot proceed with deletion")
+		return err
+	}
+
+	// Get the entire listener port object.
+	// Skip if 404, otherwise error
+	lbPort, err := e.getLoadBalancerPort(ctx, lbID, int32(portNumber))
+	if err != nil {
+		log.Error(err, "failed to load the loadbalancer port object, cannot proceed with deletion")
+		return err
+	}
+
+	resp, err := e.deleteListenerPort(ctx, lbPort.GetId())
```

Comment:
IIRC, the LB ports are automatically deleted when the LB is deleted.  The only time we need to explicitly delete the port is if we're keeping the LB and need to change which ports are open.

Reply:
Confirmed. Going to remove this.